lib/beaker/dsl/helpers/puppet_helpers.rb

```diff
@@ -300,8 +300,18 @@ def puppet_conf_for host, conf_opts
           new_conf
         end
 
+        # Restarts the named puppet service
+        #
+        # @param [Host] host Host the service runs on
+        # @param [String] service Name of the service to restart
+        # @param [Fixnum] curl_retries Number of times to retry the restart command
+        # @param [Fixnum] port Port to check status at
+        #
+        # @return [Result] Result of last status check
         # @!visibility private
-        def bounce_service host, service, curl_retries = 120
+        def bounce_service host, service, curl_retries = nil, port = nil
+          curl_retries = 120 if curl_retries.nil?
```

Comment:
yes, that is the desired case, so that we can set `port` without needing to know/update `curl_retries` later.

I'm down to use better patterns when we can, so I'd like to have this conversation whenever you can have it.